azurelinuxagent/common/protocol/util.py

```diff
@@ -317,34 +262,33 @@ def clear_protocol(self):
                 return
             logger.error("Failed to clear protocol endpoint: {0}", e)
 
-    def get_protocol(self, by_file=False):
+    def get_protocol(self):
         """
-        Detect protocol by endpoints, if by_file is True,
-        detect MetadataProtocol in priority.
+        Detect protocol by endpoint.
         :returns: protocol instance
         """
-
         if self._protocol is not None:
             return self._protocol
 
-        try:
-            self._protocol = self._get_protocol()
+        # If metadataserver certificates is present we clean certificates
+        # and remove MetadataServer firewall rule. It is possible
+        # Wire Protocol is being used due to a previous intermediate upgrade
+        # before 2.2.47 but metadata artifacts were not cleaned up.
+        if is_metadata_server_artifact_present():
+            cleanup_metadata_server_artifacts(self.osutil)
+
+        protocol_file_path = self._get_protocol_file_path()
+        if os.path.isfile(protocol_file_path) and fileutil.read_file(protocol_file_path) == WIRE_PROTOCOL_NAME:
```

Comment:
do you have an scenario where the metadata cert is not present, but the protocol file still contains "Metadata"?

if not, i think we can remove "and fileutil.read_file(protocol_file_path) == WIRE_PROTOCOL_NAME" here.

Reply:
This will break the migration scenario from MDS to WireServer because we will return early and not generate WireServer certificates here. If we don't check the file contents and the file contains MetadataServer, we will not do _detect_protocol which takes care of generating the WireServer certificates.

Reply:
it's ok now, the comment you added addresses my question